apps/stripe/src/app/api/saleor/transaction-initialize-session/use-case.ts

```diff
@@ -0,0 +1,102 @@
+import { buildSyncWebhookResponsePayload } from "@saleor/app-sdk/handlers/shared";
+import { err, ok } from "neverthrow";
+
+import { TransactionInitializeSessionEventFragment } from "@/generated/graphql";
+import { BaseError } from "@/lib/errors";
+import { createLogger } from "@/lib/logger";
+import { AppConfigPersistor } from "@/modules/app-config/app-config-persistor";
+import { SaleorApiUrl } from "@/modules/saleor/saleor-api-url";
+import { IStripePaymentIntentsApiFactory } from "@/modules/stripe/types";
+
+export class InitializeStripeTransactionUseCase {
+  private logger = createLogger("InitializeStripeTransactionUseCase");
+  private configPersister: AppConfigPersistor;
+  private stripePaymentIntentsApiFactory: IStripePaymentIntentsApiFactory;
+
+  static UseCaseError = BaseError.subclass("InitializeStripeTransactionUseCaseError");
+  static MissingConfigError = this.UseCaseError.subclass("MissingConfigError");
+
+  constructor(deps: {
+    configPersister: AppConfigPersistor;
+    stripePaymentIntentsApiFactory: IStripePaymentIntentsApiFactory;
+  }) {
+    this.configPersister = deps.configPersister;
+    this.stripePaymentIntentsApiFactory = deps.stripePaymentIntentsApiFactory;
+  }
+
+  async execute(args: {
+    channelId: string;
+    appId: string;
+    saleorApiUrl: SaleorApiUrl;
+    event: TransactionInitializeSessionEventFragment;
+  }) {
+    const { channelId, appId, saleorApiUrl } = args;
+
+    const stripeConfigForThisChannel = await this.configPersister.getStripeConfig({
+      channelId,
+      appId,
+      saleorApiUrl,
+    });
+
+    if (stripeConfigForThisChannel.isErr()) {
+      return err(
+        new InitializeStripeTransactionUseCase.UseCaseError(
```

Comment:
Idea: What about we extract some shared errors? For example, every use case will check for config, do we need to re-declare errors? Maybe just create shared ones, and maybe just add prop with cause?